src/Board/index.jsx

```diff
@@ -0,0 +1,215 @@
+import styles from "./board.module.css";
+import Row from "../Row";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+import { BsArrowRightShort } from "react-icons/bs";
+
+const GameStatus = {
+  draw: "draw",
+  inProgress: "inProgress",
+  gameOver: "gameOver",
+};
+
+const Turn = {
+  playerOne: "playerOne",
+  playerTwo: "playerTwo",
+};
+
+const H1Title = {
+  game: "Play:",
+  playerWon: `Player Won:`,
+  draw: "Draw!",
+};
+
+const initializeField = () => {
+  return {
+    id: uuid(),
+    value: null,
+    disabled: false,
+  };
+};
+
+const initialBoard = {
+  firstRow: [initializeField(), initializeField(), initializeField()],
+  secondRow: [initializeField(), initializeField(), initializeField()],
+  thirdRow: [initializeField(), initializeField(), initializeField()],
+};
+
+const Board = ({ firstPlayerName, secondPlayerName }) => {
+  const [board, setBoard] = useState(initialBoard);
+  const [gameStatus, setGameStatus] = useState(GameStatus.inProgress);
+  const [turn, setTurn] = useState(Turn.playerOne);
+  const [h1Title, setH1Title] = useState(H1Title.game);
+  const [arrowFirstPlayerClass, setArrowFirstPlayerClass] = useState(
+    styles.arrowForPlayerOn,
+  );
+  const [arrowSecondPlayerClass, setArrowSecondPlayerClass] = useState(
+    styles.arrowForPlayerOff,
+  );
+
+  const checkIfWonOrDrawAndSetGameStatus = () => {
+    checkIfWonHorizontally();
+    checkIfWonVertically();
+    checkIfWonDiagonally();
+    checkIfDraw();
+  };
+
+  const checkConditionsToWin = (array) => {
+    return allEqual(array) && array[0].value !== null;
+  };
+  const allEqual = (array) =>
+    array.every((object) => object.value === array[0].value);
+
+  const checkIfWonHorizontally = () => {
+    for (const row of Object.values(board)) {
+      if (checkConditionsToWin(row)) {
+        setGameStatus(GameStatus.gameOver);
+        setH1Title(H1Title.playerWon);
+        showWinner();
+      }
+    }
+  };
+
+  const checkIfWonVertically = () => {
+    for (let i = 0; i < Object.keys(board).length; i++) {
+      let horizontalResultArray = [];
+      for (const row of Object.values(board)) {
```

Comment:
Please make this code more readable by introducing new variables/constans or splitting it into well-named functions.

Reply:
done